src/ray/object_manager/object_manager.cc

```diff
@@ -90,9 +88,7 @@ ObjectManager::ObjectManager(
           [this, delete_object_callback = std::move(delete_object_callback)](
               const ObjectID &object_id) {
             main_service_->post(
-                [this,
-                 object_id,
-                 delete_object_callback = std::move(delete_object_callback)]() {
+                [this, object_id, &delete_object_callback]() {
```

Comment:
Hmm isn't this GC'ed when it is posted to main service though?

Reply:
It'll. But I think we call io_context.stop() first. so the task won't run.